src/ring_theory/roots_of_unity.lean

```diff
@@ -433,6 +433,9 @@ variables {M₀ : Type*} [comm_monoid_with_zero M₀]
 lemma zero [nontrivial M₀] : is_primitive_root (0 : M₀) 0 :=
 ⟨pow_zero 0, λ l hl, by simpa [zero_pow_eq, show ∀ p, ¬p → false ↔ p, from @not_not] using hl⟩
 
+lemma ne_zero' [nontrivial M₀] {ζ : M₀} {n : ℕ} (h : is_primitive_root ζ n) : n ≠ 0 → ζ ≠ 0 :=
```

Comment:
```suggestion
lemma ne_zero [nontrivial M₀] {ζ : M₀} {n : ℕ} (h : is_primitive_root ζ n) (n ≠ 0) : ζ ≠ 0 :=
```
Can you please check if there is already a variable `{k : ℕ}` floating around?

Reply:
We can't have the `'`, else whenever we use `ne_zero` in any flt-regular adjacent stuff, we instead need to use `_root_.ne_zero`. Any other names would be welcome :)

Reply:
We can maybe make the lemma `protected`? In any case it's better to have `(h : k ≠ 0)` as an assumption.

Reply:
I thought it didn't matter - what's the difference? I wanted it as an implication mainly because it golfs the proof a little

Reply:
It's of course the same thing, but in the preferred style in mathlib is (I think) to have `(h : k ≠ 0)`.

Reply:
https://leanprover-community.github.io/contribute/style.html#hypotheses-left-of-colon

Reply:
that's only if I'm introducing it straight away, which I'm not:

> if the proof starts by introducing these variables

another example of something like this is [`left_ne_zero_of_mul`](https://github.com/leanprover-community/mathlib/blob/b7cba57674fc37bca24fa9f3da3939db6e434ddc/src/algebra/group_with_zero/basic.lean#L76)

Reply:
Ah, you're right, sorry.